src/application/redux/selectors/wallet.selector.ts

```diff
@@ -172,3 +200,56 @@ export function selectRestorerOpts<T extends StateRestorerOpts>(account: Account
     return state.wallet.accounts[account].restorerOpts[net];
   };
 }
+
+export type History = Array<{ height: number; tx_hash: string }>;
+
+// take an history as input and compare it to the transactions already in the state
+// returns the txids of the new transactions (not in the state) and the txids of the transactions that are confirmed (in the state but not in a block)
+export function selectHistoryDiff(history: History) {
+  return (
+    state: RootReducerState
+  ): {
+    newTxs: { txID: string; height: number }[];
+    confirmedTxs: { txID: string; height: number }[];
+  } => {
+    const newTxs: { txID: string; height: number }[] = [];
+    const confirmedTxs: { txID: string; height: number }[] = [];
+
+    for (const { tx_hash, height } of history) {
+      const txInState = selectTransactionByID(tx_hash)(state);
```

Comment:
if you just keep in chrome.storage the outpoint, any lookup it's much more easier